packages/lodestar/test/e2e/singleNodeSingleThread.test.ts

```diff
@@ -0,0 +1,53 @@
+import {IBeaconParams} from "@chainsafe/lodestar-params";
+import {getDevBeaconNode} from "../utils/node/beacon";
+import {waitForEvent} from "../utils/events/resolver";
+import {Checkpoint} from "@chainsafe/lodestar-types";
+import {getDevValidators} from "../utils/node/validator";
+import { expect } from "chai";
+
+describe("Run single node single thread interop validators (no eth1) until checkpoint", function () {
+  const timeout = 120 * 1000;
+  const testParams: Partial<IBeaconParams> = {
+    SECONDS_PER_SLOT: 2,
+    SLOTS_PER_EPOCH: 8
+  };
+  const manyValidatorParams: Partial<IBeaconParams> = {
+    ...testParams,
+    TARGET_AGGREGATORS_PER_COMMITTEE: 1
+  };
+
+  const testCases: {
+    vc: number;
+    validators: number;
+    event: "justifiedCheckpoint" | "finalizedCheckpoint";
+    params: Partial<IBeaconParams>;
+  }[] = [
+    {vc: 8, validators: 8, event: "justifiedCheckpoint", params: testParams},
+    {vc: 8, validators: 8, event: "finalizedCheckpoint", params: testParams},
+    {vc: 1, validators: 32, event: "justifiedCheckpoint", params: manyValidatorParams},
+  ];
+
+  for (const testCase of testCases) {
+    it(`${testCase.vc} vc / ${testCase.validators} validator > until ${testCase.event}`, async function () {
+      this.timeout(timeout);
+      const bn = await getDevBeaconNode({
+        params: testCase.params,
+        options: {sync: {minPeers: 0}},
+        validatorCount: testCase.vc * testCase.validators
+      });
+      const justificationEventListener = waitForEvent<Checkpoint>(bn.chain, testCase.event, timeout - 10 * 1000);
+      const validators = getDevValidators(bn, testCase.vc, testCase.validators);
+      await bn.start();
+      await Promise.all(validators.map(v => v.start()));
+      try {
+        await justificationEventListener;
+      } catch (e) {
+        await Promise.all(validators.map((v) => v.stop()));
+        await bn.stop();
+        expect(`failed to get event: ${testCase.event}`);
+      }
+      await Promise.all(validators.map((v) => v.stop()));
+      await bn.stop();
```

Comment:
Maybe you can use a similar structure to 
https://github.com/ChainSafe/lodestar/blob/87203cca389ede5361538744975be2a368af9e81/packages/lodestar/test/e2e/multiNodeSingleThread.test.ts#L18
to prevent duplicating cleaning code and make use of Mocha's after block.

Reply:
I just tried this using the same pattern and it didn't work. Ended up running out of memory.

Reply:
seems it wasn't the pattern at fault, for the OOM, but it doesn't clear the nodes appropriately